code/core/src/core-server/presets/common-preset.ts

```diff
@@ -104,8 +104,32 @@ export const favicon = async (
 
 export const babel = async (_: unknown, options: Options) => {
   const { presets } = options;
-
-  return presets.apply('babelDefault', {}, options);
+  const babelDefault = await presets.apply('babelDefault', {}, options);
```

Comment:
Following the conversation [here](https://github.com/babel/babel/discussions/13826), using overrides mean, that our definition of @babel/preset-env with its configuration is taken for all story files and if the user has a @babel/preser-env preset in place with a custom configuration, it will get ignored for story files. Can we be sure that all other configuration options of @babel/preset-env are therefore irrelevant or do we have to make sure to try to read the users config definition of @babel/preset-env and apply all settings except of the bugfixes and target option?

Reply:
Yeah, I have read through all the options:

[targets](https://babeljs.io/docs/babel-preset-env#targets)
[bugfixes](https://babeljs.io/docs/babel-preset-env#bugfixes)
[spec](https://babeljs.io/docs/babel-preset-env#spec)
[loose](https://babeljs.io/docs/babel-preset-env#loose)
[modules](https://babeljs.io/docs/babel-preset-env#modules)
[debug](https://babeljs.io/docs/babel-preset-env#debug)
[include](https://babeljs.io/docs/babel-preset-env#include)
[exclude](https://babeljs.io/docs/babel-preset-env#exclude)
[useBuiltIns](https://babeljs.io/docs/babel-preset-env#usebuiltins)
[corejs](https://babeljs.io/docs/babel-preset-env#corejs)
[forceAllTransforms](https://babeljs.io/docs/babel-preset-env#forcealltransforms)
[configPath](https://babeljs.io/docs/babel-preset-env#configpath)
[ignoreBrowserslistConfig](https://babeljs.io/docs/babel-preset-env#ignorebrowserslistconfig)
[browserslistEnv](https://babeljs.io/docs/babel-preset-env#browserslistenv)
[shippedProposals](https://babeljs.io/docs/babel-preset-env#shippedproposals)

But none of them seem really relevant to inherit, and many are really legacy stuff. 

In theory, I actually would love it to inherit the target config if it is *more* modern than the browsers we support by default.

But because getting the user env options is quite hairy, I kind of like it much better that we just use a well-supported `override` babel API than doing any kind of merging ourselves.

I think that once Babel 8 comes out, we should revert this PR. As the defaults for babel 8 will be `target: defaults` and `bugfixes: true`, which does support async/await and destructuring out of the box.

Of course the user could still disable it, but then it needs to set a browser list lower than:
chrome 55
safari 11
firefox 54
edge 15

And I think that is quite an edge case, and fine to not support fixtures for those kind of cases.